doc: Improve "Features of the nRF91 Series" documentation (#13354)
nrfconnect/sdk-nrf: doc/nrf/device_guides/working_with_nrf/nrf91/nrf91_features.rst

  Delta patches are applied as firmware over-the-air (FOTA) upgrades.
  See :ref:`nrf9160_fota` for more information.

.. _nrf9160_ug_band_lock:

Band lock
=========

The modem can operate on a number of LTE bands.
To check which bands are supported by a particular modem firmware version, see the release notes for that version.

You can use band lock to restrict modem operation to a subset of the supported bands, which might improve the performance of your application.
To check which bands are certified in your region, visit `nRF9160 Certifications`_.

> To set the LTE band lock, enable the :ref:`lte_lc_readme` library in your project configuration file :file:`prj.conf`, by setting the Kconfig option :kconfig:option:`CONFIG_LTE_LINK_CONTROL`  to ``y``.

Reviewer: ```suggestion
To set the LTE band lock, enable the :ref:`lte_lc_readme` library by setting the Kconfig option :kconfig:option:`CONFIG_LTE_LINK_CONTROL`  to ``y`` in your :file:`prj.conf` project configuration file.
```

> Then, enable the LTE band lock feature and the band lock mask in the configuration file of your project, as follows::

Reviewer: ```suggestion
Then, enable the LTE band lock feature and the band lock mask in the project configuration file, as follows::
```

To perform a FOTA upgrade, complete the following steps:

1. Make sure that your application supports FOTA upgrades.

   To download and apply FOTA upgrades, your application must use the :ref:`lib_fota_download` library.

>    This library deduces the type of upgrade by inspecting the header of the firmware and invokes the :ref:`lib_dfu_target` library to apply the firmware upgrade.

Reviewer: ```suggestion
   This library determines the type of upgrade by inspecting the header of the firmware and invokes the :ref:`lib_dfu_target` library to apply the firmware upgrade.
```

   In its default configuration, the DFU target library is set to support all the types of FOTA upgrades except full modem firmware upgrades, but you can freely enable or disable the support for specific targets.
   In addition, the following requirements apply:

>    * To upgrade the application, you must use :doc:`mcuboot:index-ncs` as the upgradable bootloader (:kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` must be enabled).

Reviewer: ```suggestion
   * To upgrade the application, you must use :ref:`MCUboot <mcuboot:index-ncs>` as the upgradable bootloader (:kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` must be enabled).
```

>    * If you want to upgrade the upgradable bootloader, the :ref:`bootloader` must be used (:kconfig:option:`CONFIG_SECURE_BOOT`).

Reviewer: ```suggestion
   * If you want to upgrade the upgradable bootloader, you must use the :ref:`bootloader` (:kconfig:option:`CONFIG_SECURE_BOOT must be enabled`).
```
To match the wording of the previous bullet.

>    * If you want to upgrade the modem firmware through modem delta updates, neither MCUboot nor the immutable bootloader are required, because the modem firmware upgrade is handled by the modem itself.

Reviewer: ```suggestion
   * If you want to upgrade the modem firmware through modem delta updates, you don't need to use MCUboot or the immutable bootloader, because the modem firmware upgrade is handled by the modem itself.
```

   * If you want to perform a full modem firmware upgrade, an |external_flash_size| is required.

#. Create a binary file that contains the new image.

   .. note::
      This step does not apply for upgrades of the modem firmware.
      You can download delta patches and full binaries of the modem firmware from the `nRF9160 product website (compatible downloads)`_.

>    fota_upgrades_building|

Reviewer 2: ```suggestion
   |fota_upgrades_building|
```

* The :ref:`gnss_sample` sample does not use assistance by default but can be configured to use nRF Cloud A-GNSS, P-GPS, or a combination of both.
  The sample displays tracking and fix information as well as NMEA strings in the serial console.

.. _nrf9160_ug_band_lock:

Band lock
*********

The modem can operate on a number of LTE bands.

> To check which bands are supported by a particular modem firmware version, see the `nRF9160 product website (compatible downloads)`_.

Reviewer: Do you know which part of the linked section has the information? I couldn't find anything on a quick look, so I would guess the same holds true for our users as well. Might as well improve the information on this part as well while we're working on the page.

Author: The supported bands are listed in modem firmware release notes, which are inside the zip files. It's not possible to link directly to the information, but we can improve the text to be more specific. I'll try to come up with something.

Each bit in the :kconfig:option:`CONFIG_LTE_LOCK_BAND_MASK` option represents one band.
The maximum length of the string is 88 characters (bit string, 88 bits).

For more detailed information, see the `band lock section in the AT Commands reference document`_.

.. _nrf9160_ug_network_mode:

System mode
***********

> The modem system mode configuration is used to select which of the supported systems, :term:`LTE-M`, :term:`NB-IoT<Narrowband Internet of Things (NB-IoT)>` and :term:`GNSS<Global Navigation Satellite System (GNSS)>`, are enabled.

Reviewer: Either
```suggestion
You can configure the system mode of the modem to select which of the supported systems, :term:`LTE-M`, :term:`NB-IoT<Narrowband Internet of Things (NB-IoT)>` and :term:`GNSS<Global Navigation Satellite System (GNSS)>`, are enabled.
```
Unless "system mode configuration" is a thing in itself, in which case:
```suggestion
The system mode configuration of the modem is used to select which of the supported systems, :term:`LTE-M`, :term:`NB-IoT<Narrowband Internet of Things (NB-IoT)>` and :term:`GNSS<Global Navigation Satellite System (GNSS)>`, are enabled.
```

Author: The latter one sounds better to me, I'll use that.

> When using the LTE link control library, by default all supported systems are enabled and modem selects the used LTE system based on the LTE system mode preference.

Reviewer 2: ```suggestion
When using the :ref:`lte_lc_readme` library, all supported systems are enabled by default and the modem selects the used LTE system based on the LTE system mode preference.
```

You can change the enabled systems using the :kconfig:option:`CONFIG_LTE_NETWORK_MODE` Kconfig option and the LTE system mode preference using the :kconfig:option:`CONFIG_LTE_MODE_PREFERENCE` Kconfig option.

> When LTE link control library is not used, the modem starts in LTE-M mode.

Reviewer 2: ```suggestion
When the :ref:`lte_lc_readme` library is not used, the modem starts in LTE-M mode.
```
